src/Repository/Pap/CampaignRepository.php

```diff
@@ -62,13 +62,60 @@ public function countActiveCampaign(): int
 
     public function findActiveCampaignsVotePlaceIds(): array
     {
-        return array_column($this->createQueryBuilder('campaign')
-            ->select('DISTINCT votePlace.id')
+        $activeCampaign = $this->createQueryBuilder('campaign')
             ->innerJoin('campaign.votePlaces', 'votePlace')
             ->where('campaign.finishAt IS NOT NULL AND campaign.finishAt > :now')
             ->setParameter('now', new \DateTime())
             ->getQuery()
-            ->getArrayResult(), 'id'
+            ->getSingleResult()
+        ;
+
+        $queryBuilder = $this->createQueryBuilder('campaign')
+            ->select('DISTINCT votePlace.id')
+            ->innerJoin('campaign.votePlaces', 'votePlace')
+            ->where('campaign.finishAt IS NOT NULL AND campaign.finishAt > :now')
+            ->setParameter('now', new \DateTime())
+        ;
+
+        if ($activeCampaign) {
+            if ($deltaPredictionAndResultMin2017 = $activeCampaign->getDeltaPredictionAndResultMin2017()) {
+                $queryBuilder->andWhere(sprintf('votePlace.deltaPredictionAndResult2017 >= %s', $deltaPredictionAndResultMin2017));
```

Comment:
```suggestion
                $queryBuilder->andWhere('votePlace.deltaPredictionAndResult2017 >= :deltaPredictionAndResultMin2017');
                $queryBuilder->setParameter('deltaPredictionAndResultMin2017', $deltaPredictionAndResultMin2017);
```

same for other filters